src/ts/state/reducer.ts

```diff
@@ -463,11 +463,12 @@ const handleNewDirtyRanges = <TPluginState extends IPluginState>(
       )
     : state.decorations;
 
-  // Remove any matches and associated decorations
-  // touched by the dirtied ranges from the doc
+  // Remove any matches and associated decorations touched by the dirtied ranges from the doc
+  // We are providing a from offset of -1 as the range provides a cursor position,
+  // and we need to ensure that the range includes the cursor position before it.
   newDecorations = removeDecorationsFromRanges(newDecorations, dirtiedRanges);
-  const currentMatches = state.currentMatches.filter(
-    output => findOverlappingRangeIndex(output, dirtiedRanges) === -1
+  const currentMatches = state.currentMatches.filter(output =>
+    findOverlappingRangeIndex(output, dirtiedRanges, -1)
```

Comment:
This works – but possibly by coincidence! `findOverlappingRangeIndex` returns an index, which, if there's a single range dirtied, _happens_ to always to be zero (and falsy), and thus this filter behaviour works ... but are we sure that we'll always dirty a single range in a transaction?

My suspicion is that this will break if there are multiple ranges (e.g. on document cleanup (the wand icon in Composer), so I'd recommend being explicit with a comparison to an absent index (`!== -1`).

Reply:
Oops yes I think the comparison got knocked out during the edit!